junit-jupiter-params/src/test/java/org/junit/jupiter/params/ParameterizedTestExtensionTests.java

```diff
@@ -228,4 +244,21 @@ void method(String parameter) {
 		}
 	}
 
+	static class TestCaseWithArgumentSourceAnnotatedMethodWithNonStaticArgumentProvider {
+
+		@ParameterizedTest
+		@ArgumentsSource(NonStaticArgumentProvider.class)
+		void method(Collection<String> parameters) {
+		}
+
+	}
+
+	class NonStaticArgumentProvider implements ArgumentsProvider {
+
+		@Override
+		public Stream<? extends Arguments> provideArguments(ExtensionContext context) throws Exception {
+			return Stream.of(Collections.singletonList("foo")).map(Arguments::of);
```

Comment:
There's no need to return anything meaningful if the method will never be invoked. So feel free to simply return `null` here.